### ISSUE TITLE
MSW: wxTextCtrl SetStyle() font size does not take into account display scale factor

### ISSUE THREAD

**Issue by willco007** [wxWidgets/wxWidgets#22542]
**Describe the bug**
When calling `SetStyle()` on `wxTextCtrl` with a font that is 10 pt, the font size ends up being 5 pt on screen when using HiDPI (I'm on a 4k monitor). I did not test other platforms.

**Expected vs observed behaviour**
The scale factor is taken into account when calling `SetStyle()`.  Actual results are it's not.

**Patch or snippet allowing to reproduce the problem**

font here is a 10 pt font that was selected from the `wxFontPickerCtrl` and passed directly into this control.

`log_ctrl->SetStyle(0, log_ctrl->GetValue().Length(), wxTextAttr(wxColor("#fff"), wxNullColour, font));`

**Platform and version information**
 - wxWidgets version you use: 3.1.5
 - wxWidgets port you use: wxMSW
 - OS and its version: Windows 11 22000.675

**Comment by willco007** [comment]
This work-around fixes the issue if you call it before passing the font into `SetStyle()`

`font.SetPointSize(font.GetPointSize());`

**Comment by vadz** [comment]
Could you please check 3.1.7 or master? I think this might have been fixed since 3.1.5, at least there were several changes to the font scale handling since then and right now
```diff
diff --git a/samples/text/text.cpp b/samples/text/text.cpp
index 4d40c8dcfb..089e2abd7b 100644
--- a/samples/text/text.cpp
+++ b/samples/text/text.cpp
@@ -1248,8 +1248,9 @@ void MyTextCtrl::OnKeyDown(wxKeyEvent& event)
                                 wxTE_RICH | wxTE_MULTILINE | wxTE_AUTO_URL);
     m_textrich->SetStyle(0, 10, *wxRED);
     m_textrich->SetStyle(10, 20, *wxBLUE);
+    wxFont font(wxFontInfo(10).Underlined());
     m_textrich->SetStyle(30, 40,
-                         wxTextAttr(*wxGREEN, wxNullColour, *wxITALIC_FONT));
+                         wxTextAttr(*wxGREEN, wxNullColour, font));
     m_textrich->SetDefaultStyle(wxTextAttr());
     m_textrich->AppendText("\n\nFirst 10 characters should be in red\n");
     m_textrich->AppendText("Next 10 characters should be in blue\n");
```
works correctly for me. TIA!

**Comment by willco007** [comment]
Finally updated our wxWidgets to 3.2.x and I can no longer reproduce the issue. Looks like it was fixed.
